### PR TITLE
Reduce vocabulary size in test model, fix bug in routing when overlapped

### PR DESCRIPTION
**Pull request by justheuristic**
More than half of the test 560m model parameters is it's embedding layer (250_880 x 1024).
This adds 1-2 GB of RAM depending on the test scenario
This PR reduces this vocabulary size to save memory during conversion, keeping only the first 50k tokens

As a result,
* tests that load client-side embeddings need significantly less RAM
* we can now run CI tests with 4 servers instead of 2 - needed to test routing - see bugs uncovered
* some of the servers now use load balancing
* CI convert_model now takes 4-5 minutes (was 6-7)

Running CI with 4 servers uncovered a bug:
https://github.com/bigscience-workshop/petals/runs/7879399020?check_suite_focus=true
which is now also fixed

This PR also temporarily pins an older bnb version, pending @TimDettmers 's fix for cpuonly

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -26,6 +26,11 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
+      - name: Delete test models older than 72 hours
+        run: |
+          export HF_TAG=$(python -c "import os; print(os.environ.get('GITHUB_HEAD_REF') or os.environ.get('GITHUB_REF_NAME'))")
+          python -c "from huggingface_hub import delete_repo; delete_repo(token='$BLOOM_TESTING_WRITE_TOKEN', \
+          name='test-bloomd-560m-$HF_TAG', organization='bloom-testing')" || true
```
Comment (borzunov) [inline .github/workflows/run-tests.yaml:33]: Please consider moving that to a small Python script in the repo. Having a Bash command running Python, whose results are passed to another Bash command running Python, is not good.

Comment (borzunov) [inline .github/workflows/run-tests.yaml:33]: Please consider moving that to a small Python script in the repo. Having a Bash command running Python, whose results are passed to another Bash command running Python, is not good.

File: .github/workflows/run-tests.yaml [hunk 2]
```diff
@@ -76,21 +82,41 @@ jobs:
...
           
-          python -m cli.run_server --converted_model_name_or_path $MODEL_NAME --block_indices 12:24 \
+          python -m cli.run_server --converted_model_name_or_path $MODEL_NAME --block_indices 12:22 \
             --torch_dtype float32 --initial_peers $INITIAL_PEERS --throughput 1 &> server2.log &
           SERVER2_PID=$!
 
-          sleep 60  # wait for server to download layers
+          sleep 10 # wait for initial servers to declare blocks, then let server decide which blocks to serve
+
+          python -m cli.run_server --converted_model_name_or_path $MODEL_NAME --block_indices 0:6 \
```
Comment (borzunov) [inline .github/workflows/run-tests.yaml:100]: Please consider running a loop instead, maybe using just `--num_blocks` without explicit `--block_indices`. If it gets more complicated than just repeating something N times, please move it to a Python script.

File: .github/workflows/run-tests.yaml (hunk 1 above)
Comment (justheuristic) [reply .github/workflows/run-tests.yaml]: done, see tests/srcipts/remove_old_models.py

File: .github/workflows/run-tests.yaml (hunk 2 above)
Comment (justheuristic) [reply .github/workflows/run-tests.yaml:98]: moved to #16

**Pull request merged**